themes/bootstrap3/templates/cart/form-record-hidden-inputs.phtml

```diff
@@ -0,0 +1,24 @@
+<?php
+  $recordId = $this->driver?->getUniqueId() ?? $this->recordId ?? false;
+  $recordSource = $this->driver?->getSourceIdentifier() ?? $this->recordSource ?? DEFAULT_SEARCH_BACKEND;
+  if ($recordId && str_contains($recordId, '|')) {
```

Comment:
I'm concerned that this could malfunction in multiple ways if an ID contains a pipe character. Is there a safer way to determine when unpacking is required? i.e. do we only unpack recordId if there is no driver and no source set? (I didn't analyze all the code paths to figure out options, but I can try to spend a little time on that if it would help).

Additionally, are the variables reversed? Down below, you construct "source|id" strings, but here you're unpacking it as "id|source".

Reply:
oh, missed that... will adjust them also!

Reply:
Hmm, i think it would be safer to leave it out and let the calling templates to give what is needed?

Reply:
Yes, definitely safer this way -- thanks!